### DISCUSSION TITLE
Build vs generate behavior

### DISCUSSION THREAD

**Discussion by oscarlevin** [PreTeXtBook/pretext-cli#468] [General]
In the next stable release, the plan is that on `pretext build` the cli will also generate all needed assets for the specified target.  These will then be tracked with a hash table so that only those assets that are modified in source will be regenerated on the subsequent builds.  On one hand, this will increase efficiency quite a bit for large projects, but each target is tracked separately (which is a good idea because different targets might have different source and format, so have different assets to generate).  That means that, for example, a `webwork-representations.xml` file will be generated multiple times.  Maybe we can improve this by using multiple hash files.

Anyway, the point of this discussion is to make sure that this workflow still works for everyone, especially @bnmnetp for Runestone.  I think the assumption right now is that all generated assets are build and tracked in the generated-assets folder, which can still happen, but would not by default.  That would mean that these assets would be rebuilt (and the necessary software would need to exist in the build environment).

Perhaps this needs to be part of a larger discussion about the Runestone build process.  Thoughts?

**Comment by bnmnetp** [comment]
Thanks Oscar, it is very cool to see this happening.

So, I just want to clarify what you said about the generated-assets folder.  -- generated assets will still go there right?  We won't have to change everyone's project or publisher files to find them somewhere else right?

**Reply by oscarlevin** [reply]
Correct.  Folder structure is exactly the same (except now there is a new `target_assets.pkl` file for each target).  And if an author removed `generated-assets` from their `.gitignore`, that would be tracked too.
